operators/pkg/controller/elasticsearch/certificates/transport/reconcile_test.go

```diff
@@ -271,99 +246,59 @@ func Test_doReconcileTransportCertificateSecret(t *testing.T) {
 			secret: corev1.Secret{
 				ObjectMeta: objMeta,
 				Data: map[string][]byte{
 					certificates.CertFileName: pemCert,
-					certificates.CSRFileName:  testCSRBytes,
 					certificates.CAFileName:   certificates.EncodePEMCert(testCA.Cert.Raw),
 				},
 			},
 			additionalTrustedCAsPemEncoded:  additionalCA,
-			pod:                             podWithRunningCertInitializer,
+			pod:                             testPod,
 			wantSecretUpdated:               true,
 			wantCertUpdateAnnotationUpdated: false,
 		},
 		{
 			name:                            "no cert generated yet: issue one",
 			secret:                          corev1.Secret{ObjectMeta: objMeta},
-			pod:                             podWithRunningCertInitializer,
-			wantSecretUpdated:               true,
-			wantCertUpdateAnnotationUpdated: true,
-		},
-		{
-			name:                            "no cert generated, but pod has no IP yet: requeue",
-			secret:                          corev1.Secret{ObjectMeta: objMeta},
-			pod:                             corev1.Pod{},
-			wantSecretUpdated:               false,
-			wantCertUpdateAnnotationUpdated: true,
-		},
-		{
-			name:                            "no cert generated, but cert-initializer not running: requeue",
-			secret:                          corev1.Secret{ObjectMeta: objMeta},
-			pod:                             podWithTerminatedCertInitializer,
-			wantSecretUpdated:               false,
-			wantCertUpdateAnnotationUpdated: true,
-		},
-		{
-			name: "a cert already exists, is valid, and cert-initializer is not running: requeue",
-			secret: corev1.Secret{
-				ObjectMeta: objMeta,
-				Data: map[string][]byte{
-					certificates.CertFileName: pemCert,
-				},
-			},
-			pod:                             podWithTerminatedCertInitializer,
-			wantSecretUpdated:               false,
-			wantCertUpdateAnnotationUpdated: true,
-		},
-		{
-			name: "a cert already exists, is valid, and cert-initializer is running to serve a new CSR: issue cert",
-			secret: corev1.Secret{
-				ObjectMeta: objMeta,
-				Data: map[string][]byte{
-					certificates.CertFileName: pemCert,
-				},
-			},
-			pod:                             podWithRunningCertInitializer,
+			pod:                             testPod,
 			wantSecretUpdated:               true,
 			wantCertUpdateAnnotationUpdated: true,
 		},
 		{
-			name: "a cert already exists, is valid, and cert-initializer is running to serve the same CSR: requeue",
-			secret: corev1.Secret{
-				ObjectMeta: objMeta,
-				Data: map[string][]byte{
-					certificates.CertFileName: pemCert,
-					certificates.CSRFileName:  testCSRBytes,
-					certificates.CAFileName:   certificates.EncodePEMCert(testCA.Cert.Raw),
-				},
+			name:   "no cert generated, but pod has no IP yet: requeue",
+			secret: corev1.Secret{ObjectMeta: objMeta},
+			pod:    corev1.Pod{},
+			wantErr: func(t *testing.T, err error) {
+				assert.Contains(t, err.Error(), "pod currently has no valid IP")
 			},
```

Comment:
The flow of doReconcileTransportCertificateSecret looks complex enough to maybe require a few more unit tests?
* private key there but no cert: should be updated
* cert there but invalid: should be updated